### DISCUSSION TITLE
Autogenerate LSP definitions

### DISCUSSION THREAD

**Discussion by benwaffle** [Prince781/vala-language-server#199] [Ideas]
Ideally the spec would be in JSON or something, but according to https://github.com/microsoft/language-server-protocol/issues/67 it's not yet there.

We would have to write the code generator in TS and run a typescript parser over this:
https://github.com/microsoft/vscode-languageserver-node/tree/master/protocol/src

**Comment by benwaffle** [comment]
somebody parsed the markdown - https://gist.github.com/bollwyvl/7a128978b8ae89ab02bbd5b84d07a4b7

**Comment by Prince781** [comment]
I don't know if it would be worth the extra effort to write a parser to convert JSON to Vala objects.
